lib/licenses.nix

```diff
@@ -541,6 +541,11 @@ lib.mapAttrs (n: v: v // { shortName = n; }) {
     fullName = "Notion modified LGPL";
   };
 
+  npl1 = {
+    url = "https://git.pixie.town/thufie/NPL";
+    fullName = "Non-Violent Public License v1";
```

Comment:
```suggestion
    fullName = "Non-Violent Public License v1";
    free = false;
```

From the URL:

> A Note to those concerned about "Free Software".
>
> The Non-Violent Public License intentionally does not meet the Free Software Definition because of the restrictions put in place to protect others from violence aided by the use of the copyrighted work. "Freedom 0" is explicitly violated to afford these protections and prevent unethical usage of the program to actually Free Society. This license does preserve freedoms 1-3 and can be referred to as a "three freedoms" license. Similarly to the GNU AGPL this license has the additional requirement that server-side source code modifications to public-facing server software must be published for users as well as the usual expectations for FLOSS software with the exception of restrictions on Freedom 0 for ethical purposes.

So this is nonfree software by definition, but still, should we list it as nonfree when it still gives 3 out of the 4 freedoms for free software? I don't know but I'll try to elaborate a bit.

I respect the spirit of the license itself, it does seem to mean well but I guess the most 'controversial' part (for me) is its absolute pacifist nature i.e. no bodily harm/war/incarceration would imply not even allowed for self defense.

But without even getting into any discussion of ethics, I think really what would happen in practice with respect to violations of this license is that a state would likely already have their own laws about the various restrictions in the "Ethical Restrictions on usage" section, and that the state's laws would be favored over copyright law (which is what this license falls under). So really I think this entire "Ethical Restrictions" section is probably useless (i.e. seems like the wrong domain to be in a copyright license), but I'm no lawyer and maybe law in general doesn't work like that.

But if it is a 'useless' section, does that equate to actually having [freedom 0](https://www.gnu.org/philosophy/free-sw.en.html) with this license? Meaning it would be acceptable to **not** have `free = false`? This is what I'm unsure about but I guess either way it's probably just safer to have `free = false`. (And if we do so, maybe we should also give it `unfreeRedistributable` license?)

---

PS. I wonder what free society really means to the author of the license as I've seen these absolute pacifist views in free software/culture communities before but usually they are also very anti-copyleft unlike here (for similar reasons like anti-incarceration, seeing copyleft as inherently 'violent', etc). Also the "Extra Restriction in the CNPL" seems like an anti-capitalist restriction, whereas I've always thought free society was more pro capitalism. But I guess I'm probably remembering a different idea of free society than the author of this license.

Also the license probably needs to define a lot more terms it uses to really be usable but again, I'm no lawyer so can't really suggest anything without knowing what exactly.

Reply:
also, a license addition should be done in a separate commit

Reply:
> So this is nonfree software by definition, but still, should we list it as nonfree when it still gives 3 out of the 4 freedoms for free software?

It should be listed as non-free, IMO. It’s authors clearly do not intend for it to be free, and it’s not up to us to decide that part of the license is meaningless and should be ignored.